## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [3]:
import pandas as pd
df = pd.read_csv('../data/data.csv')

In [14]:
data = df.data.to_numpy()

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [16]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [23]:
from statsmodels.stats.weightstats import ztest

In [35]:
null_mean = 20

In [68]:
ztest_Score1, p_value1 = ztest(data, value = null_mean, alternative='larger')

In [70]:
p_value1

0.8402164165464012

Значення p-value - 0.8402 (значно більше за 0,05). Це свідчить про те, що гіпотезу H1 варто відхилити і за останній місяць не вийшло значущо підняти середній чек.

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [48]:
df2 = pd.read_csv('../data/data_2.csv')
data2 = df2.data.to_numpy()

In [72]:
ztest_Score2, p_value2 = ztest(data, data2, value=0)

In [74]:
p_value2

0.9958137794593356

In [53]:
np.mean(data), np.mean(data2)

(19.37755319670441, 19.37322276512519)

Значення p-value - 0.9958 (значно більше за 0,05). Це свідчить про те, що гіпотезу H1 варто відхилити і у серпні 2022 не вдалося статистично значущо підняти середній чек у порівнянні з липнем 2022.
Просте спостереження середніх значень (19,37 у липні і 19,37 у серпні) повністю підтверджує результати z-тесту.